pydantic/validators.py

```diff
@@ -113,6 +113,13 @@ def number_size_validator(v: 'Number', field: 'Field') -> 'Number':
     return v
 
 
+def constant_validator(v: 'Any', field: 'Field') -> 'Any':
+    if field.const and v != field.default:
```

Comment:
please add a docstring explaining what this does since it's very different from the other validators.

Also you can not remove `field.const and `